lib/web/apiserver_test.go

```diff
@@ -3565,3 +3566,35 @@ type mockProxySettings struct {
 func (mock *mockProxySettings) GetProxySettings(ctx context.Context) (*webclient.ProxySettings, error) {
 	return &webclient.ProxySettings{}, nil
 }
+
+func TestIsRoleMatchError(t *testing.T) {
+	t.Parallel()
+	matches := []struct {
+		name string
+		err  error
+	}{
+		{"saml", trace.AccessDenied("unable to map attributes to role for connector: x")},
+		{"oidc", trace.AccessDenied("unable to map claims to role for connector: x")},
+		{"github", trace.BadParameter("user y does not belong to any teams configured in z connector")},
+	}
+	for _, tc := range matches {
+		t.Run(fmt.Sprintf("Accept %s", tc.name), func(t *testing.T) {
+			require.True(t, isRoleMatchError(tc.err))
+		})
+	}
+
+	nonMatches := []struct {
+		name string
+		err  error
+	}{
+		{"nil", nil},
```

Comment:
nit: use keyed struct initialization

```suggestion
		{name: "nil"},
```

same for others.